presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSource.java

```diff
@@ -66,43 +55,15 @@
     public OrcPageSource(
             OrcRecordReader recordReader,
             OrcDataSource orcDataSource,
-            List<HiveColumnHandle> columns,
-            TypeManager typeManager,
+            Map<Integer, Type> includedColumns,
```

Comment:
I don't like the usage of `Map` here.

- it's very non-obvious what map keys are
- this is not used as a map. It's just a list of pairs `(hive column index, column type)` (... or triples `(block no, hive column index, column type)` where `block no` is implied from iteration order)

A List would be better.

I know this is pre-existing, but I think we should improve (might be a follow-up).

Reply:
In https://github.com/prestosql/presto/pull/1552 you seem to be using `List<Type> types, List<Optional<Field>> fields` for similar purpose. 
I'm ok with two-list approach.

Reply:
> I don't like the usage of `Map` here.
> 
> * it's very non-obvious what map keys are
> * this is not used as a map. It's just a list of pairs `(hive column index, column type)` (... or triples `(block no, hive column index, column type)` where `block no` is implied from iteration order)
> 
> A List would be better.
> 
> I know this is pre-existing, but I think we should improve (might be a follow-up).

Your description is precise. It's essentially a list of triples. I think we can do the improvement as a follow-up.

Reply:
This is the API we already use in [`OrcRecordReader`](https://github.com/prestosql/presto/blob/f37659ee6ce907e51a75f325ed744c2fcbc459ca/presto-orc/src/main/java/io/prestosql/orc/OrcRecordReader.java#L116), so it makes sense to me.  

That said, we are going to need some better abstractions around ORC reader columns in general to deal with features like ACID, and I think we should deal with the new API with that PR instead.